File: tests_wraps/src/main/java/com/github/klee0kai/test/car/di/bestOf/both/BothCarComponent.java
package com.github.klee0kai.test.car.di.bestOf.both;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.test.car.model.Car;

import javax.inject.Named;
import javax.inject.Provider;
import java.lang.ref.WeakReference;

@Component
public interface BothCarComponent {

    BothCarModule myModule();

    @Named("blueCar")
    Provider<Car> blueCar();

    @Named("redCar")
    WeakReference<Car> redCar();

}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/di/bestOf/both/BothCarComponent.kt
package com.github.klee0kai.test.car.di.bestOf.both

import com.github.klee0kai.stone.annotations.component.*
import com.github.klee0kai.test.car.model.Car
import java.lang.ref.WeakReference
import javax.inject.Named
import javax.inject.Provider

@Component
interface BothCarComponent {
    fun myModule(): BothCarModule?

    @Named("blueCar")
    fun blueCar(): Provider<Car>

    @Named("redCar")
    fun redCar(): WeakReference<Car>
}
